modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/components/Keywords.js

```diff
@@ -38,122 +39,197 @@ export default function Keywords({currentPage, languageTag}) {
+		<>
+			<h5 className="mt-3 sheet-subtitle">
+				{Liferay.Language.get('best-keywords')}
+			</h5>
+
+			<div className="mb-3 text-secondary">
+				{Liferay.Language.get('best-keywords-description')}
+			</div>
+
+			<div className="form-group">
+				<label htmlFor="countrySelect">
+					{Liferay.Language.get('select-country')}
+				</label>
+				<ClaySelect
+					aria-label={Liferay.Language.get('select-country')}
+					className="bg-white"
+					defaultValue={currentCountry}
+					id="countrySelect"
+					onChange={handleCountrySelection}
+				>
+					{countries.map((item) => (
+						<ClaySelect.Option
+							key={item.countryCode}
+							label={item.countryName}
+							value={item.countryCode}
+						/>
+					))}
+				</ClaySelect>
+			</div>
+
+			<ClayList className="list-group-keywords-list">
 				<ClayList.Item flex>
 					<ClayList.ItemField expand>
-						<ClayList.ItemText>
-							<span className="text-secondary">
-								{Liferay.Language.get(
-									'there-are-no-best-keywords-yet'
-								)}
+						<ClayList.ItemTitle className="text-truncate-inline">
+							<span className="text-truncate">
+								{Liferay.Language.get('best-keyword')}
+								<span className="text-secondary">
+									<Hint
+										message={Liferay.Language.get(
+											'best-keyword-help'
+										)}
+										title={Liferay.Language.get(
+											'best-keyword'
+										)}
+									/>
+								</span>
 							</span>
-						</ClayList.ItemText>
+						</ClayList.ItemTitle>
+					</ClayList.ItemField>
+					<ClayList.ItemField>
+						<ClayDropDown
+							active={isDropdownOpen}
+							onActiveChange={(isActive) =>
+								setIsDropdownOpen(isActive)
+							}
+							trigger={
+								<ClayButton
+									className="px-0 text-body"
+									displayType="link"
+									small
+								>
+									<span className="font-weight-semi-bold">
+										<span className="pr-2">
+											{keywordValueType.label}
+										</span>
+										<ClayIcon symbol="caret-bottom" />
+									</span>
+								</ClayButton>
+							}
+						>
+							<ClayDropDown.ItemList>
+								{KEYWORD_VALUE_TYPE.map((valueType, index) => (
+									<ClayDropDown.Item
+										active={
+											valueType.name ===
+											keywordValueType.name
+										}
+										key={index}
+										onClick={() => {
+											setKeywordValueType(
+												KEYWORD_VALUE_TYPE.find(
+													(keywordValueType) => {
+														return (
+															keywordValueType.name ===
+															valueType.name
+														);
+													}
+												)
+											);
+											setIsDropdownOpen(false);
+										}}
+									>
+										{valueType.label}
+									</ClayDropDown.Item>
+								))}
+							</ClayDropDown.ItemList>
+						</ClayDropDown>
 					</ClayList.ItemField>
 				</ClayList.Item>
-			)}
-		</ClayList>
+				{!publishedToday &&
+					countryKeywords.length > 0 &&
+					countryKeywords.map((k) => {
```

Comment:
Same here about abbreviations.

Reply:
👍

Reply:
In this case you don't even need a name because youcould destructure:

```js
countryKeywords.map(({keyword, traffic /* etc */}) => {
   // blah
})
```

Also, note the  `length` check is unnecessary; if the length is 0, the map will (harmlessly) return an empty array.

Reply:
Of course! You're right! Thanks! 👍